internal/packager/fork.go

```diff
@@ -0,0 +1,122 @@
+package packager
+
+import (
+    "archive/tar"
+    "io"
+    "io/ioutil"
+    "os"
+    "path/filepath"
+    "strings"
+
+    "github.com/docker/app/internal"
+    "github.com/docker/app/internal/types"
+
+    "github.com/pkg/errors"
+    log "github.com/sirupsen/logrus"
+    yaml "gopkg.in/yaml.v2"
+)
+
+// Fork pulls an application and creates a local fork for the user to modify
+func Fork(originName, forkName, outputDir string, maintainers []string) error {
+    log.Debugf("Pulling latest version of package %s", originName)
+    if err := pullImage(originName); err != nil {
+        return err
+    }
+    tmpdir, err := ioutil.TempDir("", "dockerappfork_")
+    if err != nil {
+        return err
+    }
+    defer os.RemoveAll(tmpdir)
+    log.Debugf("Extracting original app data to %s", tmpdir)
+    if err = Load(originName, tmpdir); err != nil {
+        return err
+    }
+
+    // create app dir in output-dir
+    namespace, name := splitPackageName(forkName)
+    appPath := filepath.Join(outputDir, internal.DirNameFromAppName(name))
+    os.MkdirAll(appPath, 0755)
+
+    // iterate tar contents
+    imgRef, err := splitImageName(originName)
```

Comment:
err is ignored here, we should check for it

Reply:
done, and moved it up so we don't attempt to pull if the image name isn't valid.